app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,57 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  
+  def google_oauth2
+
+      auth = request.env["omniauth.auth"]
+
+      user = User.where(:provider => auth.provider, :uid => auth.uid).first
+
+      # first time login with this provider
+      unless user
+        user = User.where(:email => auth.info.email).last
+        # first time login with this provider & no user with such email
+        unless user
+
+        # add provider and uid to the last user with this email
+        else
+          user.provider = auth.provider
+          user.uid = auth.uid
+          user.save!
+        end
+
+      end
```

Comment:
It seems to me that with the current logic an existing user will be able to login via google, but a non-existing user won't be able to register and login. Will it be possible to also register a new user if one is missing?